### PR TITLE
Change code to module & add a scraper docs

### PR DESCRIPTION
**Pull request by mmxxixx**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `.env.sample` file to guide users in configuring environment variables for GitHub integration.
	- Added new command options in the README for executing the bot with specific modules.
	- Implemented `StartDoc` and `StartForum` functions for generating documentation and processing forum data.
	- Added utility functions for argument parsing and handling GitHub API requests.

- **Bug Fixes**
	- Improved folder naming consistency when saving project data.

- **Documentation**
	- Enhanced README with new sections on module options and command execution.

- **Chores**
	- Expanded `.gitignore` to include additional directories.
	- Updated package dependencies to improve project functionality.

### CODE REVIEW

File: script/docs/docs.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// console.log("done: 'list all projects in each category' working-on: 'get each project detail' . . .")
+import { ListAllFileInDocs } from "./list-all-file-github"
+import { SaveFileMdxGithub } from "./save-file-content"
+
+export async function StartDoc() {
+    console.log("Docs is starting . . .")
+
+    
+    let docs = await ListAllFileInDocs()
+    await SaveFileMdxGithub(docs)
```
Comment (wit03) [inline script/docs/docs.ts]: Try catch would be good here as coderabbitai said (better error handling), so we can directly migrate this code to Worker V2

**Review by wit03 — CHANGES_REQUESTED** [review]
Need one fix on `script/docs/docs.ts`

File: script/docs/list-all-file-github.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        for (const res of response.data.tree){
+            if(res.path.includes(".mdx")){
+                let doc: DocPath = {
+                    path: res.path,
+                    url: res.url,
+                }
+                allUrlPath.push(doc)
+            }
+        }
```
[the remark this answers is not in the document]
Comment (wit03) [reply script/docs/list-all-file-github.ts:36]: filtering before map reduce the number of iterations

[the remark this answers is not in the document]
Comment (mmxxixx) [reply script/docs/list-all-file-github.ts:36]: for ... of  
- Single Iteration: The loop iterates through the array only once

.filter then .map 
- Two Iterations: The array is iterated twice—once for filter and once for map

[the remark this answers is not in the document]
Comment (mmxxixx) [reply script/docs/list-all-file-github.ts:36]: here are some performance test 
image
```
// Function to generate a large array of test data
function generateTestData(size) {
    return Array.from({ length: size }, (_, index) => ({
      path: index % 3 === 0 ? `file${index}.mdx` : `file${index}.js`,
      url: `https://example.com/file${index}`
    }));
  }
  
  // Implementation using for...of loop
  function forOfApproach(data) {
    const allUrlPath = [];
    for (const res of data) {
      if (res.path.includes(".mdx")) {
        allUrlPath.push({ path: res.path, url: res.url });
      }
    }
    return allUrlPath;
  }
  
  // Implementation using filter and map
  function filterMapApproach(data) {
    return data
      .filter(res => res.path.includes(".mdx"))
      .map(res => ({ path: res.path, url: res.url }));
  }
  
  // Function to run performance test
  function runPerformanceTest(size, iterations) {
    const testData = generateTestData(size);
    
    let forOfTime = 0;
    let filterMapTime = 0;
  
    for (let i = 0; i < iterations; i++) {
        // Test filter and map approach
        const startFilterMap = performance.now();
        filterMapApproach(testData);
        const endFilterMap = performance.now();
        filterMapTime += endFilterMap - startFilterMap;

        // Test for...of approach
        const startForOf = performance.now();
        forOfApproach(testData);
        const endForOf = performance.now();
        forOfTime += endForOf - startForOf;
    }
  
    console.log(`Average time for for...of approach: ${forOfTime / iterations} ms`);
    console.log(`Average time for filter and map approach: ${filterMapTime / iterations} ms`);
  }
  
  // Run the test
  const arraySize = 1000000;  // Size of the test array
  const iterations = 100;     // Number of times to run each test for averaging
  
  runPerformanceTest(arraySize, iterations);
```

[the remark this answers is not in the document]
Comment (wit03) [reply script/docs/list-all-file-github.ts:36]: Ok, nice ref!

**Review by wit03 — APPROVED** [review]

**Review by wit03 — APPROVED** [review]

**Comment by wit03** [comment]
LGTM

**Pull request merged**